File: 12. Check if two strings are anagram of each other/code.cpp
#include <iostream>
#include <algorithm>
using namespace std;

bool areAnagrams(string s1, string s2) {
    sort(s1.begin(), s1.end());
    sort(s2.begin(), s2.end());
    return s1 == s2;
}

int main() {
    string input_str1, input_str2;
    cout << "Enter the first string: ";
    cin >> input_str1;
    cout << "Enter the second string: ";
    cin >> input_str2;
    cout << "Are the strings anagrams? " << (areAnagrams(input_str1, input_str2) ? "true" : "false") << endl;
    return 0;
}

File: 12. Check if two strings are anagram of each other/code.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    public static boolean areAnagrams(String s1, String s2) {
        char[] arr1 = s1.toCharArray();
        char[] arr2 = s2.toCharArray();
        Arrays.sort(arr1);
        Arrays.sort(arr2);
        return Arrays.equals(arr1, arr2);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter the first string: ");
        String inputStr1 = scanner.nextLine();
        System.out.print("Enter the second string: ");
        String inputStr2 = scanner.nextLine();
        System.out.println("Are the strings anagrams? " + areAnagrams(inputStr1, inputStr2));
    }
}
